bijection-core/src/main/scala/com/twitter/bijection/StringInjections.scala

```diff
@@ -32,9 +33,22 @@ trait StringInjections extends NumericInjections {
 
   def withEncoding(encoding: String): Injection[String, Array[Byte]] =
     new AbstractInjection[String, Array[Byte]] {
+      private[this] val decRef = {
+        val cs = Charset.forName(encoding)
+        new AtomicSharedState(() => cs.newDecoder.onUnmappableCharacter(CodingErrorAction.REPORT))
+      }
+
       def apply(s: String) = s.getBytes(encoding)
       override def invert(b: Array[Byte]) =
-        attempt(b) { new String(_, encoding) }
+        // toString on ByteBuffer is nicer, so use it
+        attempt(ByteBuffer.wrap(b)) { bb =>
```

Comment:
the attempt function includes a toSting of the argument, which in this case is the ByteBuffer.